pulseapi/entries/views.py

```diff
@@ -26,77 +29,62 @@
 from pulseapi.utility.userpermissions import is_staff_address
 
 @api_view(['PUT'])
+@permission_classes([IsAuthenticated])
 def toggle_bookmark(request, entryid):
     """
     Toggle whether or not this user "bookmarked" the url-indicated entry.
     This is currently defined outside of the entry class, as functionality
     that is technically independent of entries themselves. We might
     change this in the future.
     """
-    user = request.user
 
-    if user.is_authenticated():
-        entry = None
+    user = request.user
+    entry = get_object_or_404(Entry, id=entryid)
+    # find is there is already a {user,entry,(timestamp)} triple
+    bookmarks = entry.bookmarked_by.filter(user=user)
+    exists = bookmarks.count() > 0
+
+    # if there is a bookmark, remove it. Otherwise, make one.
+    if exists:
+        for bookmark in bookmarks:
+            bookmark.delete()
+    else:
+        bookmark = UserBookmarks(entry=entry, user=user)
+        bookmark.save()
 
-        # find the entry for this id
-        try:
-            entry = Entry.objects.get(id=entryid)
-        except Entry.DoesNotExist:
-            return Response("No such entry", status=status.HTTP_404_NOT_FOUND)
+    return Response("Toggled bookmark.", status=status.HTTP_204_NO_CONTENT)
 
-        # find is there is already a {user,entry,(timestamp)} triple
-        bookmarks = entry.bookmarked_by.filter(user=user)
-        exists = bookmarks.count() > 0
 
-        # if there is a bookmark, remove it. Otherwise, make one.
-        if exists:
-            for bookmark in bookmarks:
-                bookmark.delete()
-        else:
-            bookmark = UserBookmarks(entry=entry, user=user)
-            bookmark.save()
+@api_view(['PUT'])
+@permission_required('entries.can_change_entry', raise_exception=True)
+def toggle_featured(request, entryid):
+    """
+    Toggle the featured status of an entry.
+    """
+    entry = get_object_or_404(Entry, id=entryid)
+    entry.featured = not entry.featured
+    entry.save()
 
-        return Response("Toggled bookmark.", status=status.HTTP_204_NO_CONTENT)
-    return Response("Anonymous bookmarks cannot be saved.", status=status.HTTP_403_FORBIDDEN)
+    return Response("Toggled featured status.",
+                    status=status.HTTP_204_NO_CONTENT)
```

Comment:
I recommended using a function for this response object because it means we no longer have to break code just because it's 4 characters too long. If we're putting in a linebreak here, let's use that function call instead and improve all `return Response` lines.